chain/index/events.go

```diff
@@ -280,6 +282,39 @@ func (si *SqliteIndexer) checkTipsetIndexedStatus(ctx context.Context, f *EventF
 	return ErrNotFound // Tipset is not indexed
 }
 
+func (si *SqliteIndexer) checkRangeIndexedStatus(ctx context.Context, f *EventFilter) error {
+	startCid, err := si.getTipsetKeyCidByHeight(ctx, f.MinHeight)
+	if err != nil {
+		if errors.Is(err, ErrNotFound) {
+			// Null round for the start of the range is acceptable
+			return nil
+		}
+		return xerrors.Errorf("failed to get tipset key cid for start height: %w", err)
+	}
+
+	endCid, err := si.getTipsetKeyCidByHeight(ctx, f.MaxHeight)
+	if err != nil {
+		if errors.Is(err, ErrNotFound) {
+			// Check if the range ends in the future
+			head := si.cs.GetHeaviestTipSet()
+			if head == nil || f.MaxHeight > head.Height() {
+				return xerrors.Errorf("range end is in the future: %w", err)
+			}
+			return ErrNotFound // End is unindexed
```

Comment:
We're back in the null round case, I think that we should walk backward to find the first non-null lower than `MaxHeight`. Basically the same as the approach outlined above for `MinHeight` but in the other direction. You may find a way to abstract the algorithm to make it a reusable function, or it may be easier to just write it all out.

Basically: let's contract the range if the edges contain null rounds, walk _inward_ until we find a min and max that are not null rounds. Be careful not to iterate beyond the other bound (don't increment from Min beyond the Max value, and don't decrement from Max beyond the Min value). The special case is when there are _only_ null rounds between them, don't error, just return. There's another special case of contracting to min==max, you could easily do an `if` block with `startCid.Equals(endCid)` around the second `isTipsetIndexed` call to deal with that (although it's not critical since it'd just mean two queries for the same thing if it wasn't handled).